File: 997 - Find the Town Judge.py
class Solution:
    def findJudge(self, n, trust):
        trustor = dict()
        trusted = dict()
        for i in range(1, n + 1):
            trustor.setdefault(i, [])
            trusted.setdefault(i, [])
        for a, b in trust:
            trustor[b].append(a)
            trusted[a].append(b)
        for key in trustor.keys():
            if len(trustor[key]) == n - 1 and len(trusted[key]) == 0:
                return key
        return -1


s = Solution()
s.findJudge(3, [[1, 3], [2, 3], [3, 1]])

File: 997 - Find the Town Judge.cpp
#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
        unordered_map<int, vector<int>> trustor, trusts;
        for (int i = 1; i <= n; i++) {
            trustor[i] = vector<int>();
            trusts[i] = vector<int>();
        }
        for (int i = 0; i < trust.size(); i++) {
            int a = trust[i][0], b = trust[i][1];
            trustor[b].push_back(a);
            trusts[a].push_back(b);
        }

        for (auto item = trustor.begin(); item != trustor.end(); item++) {
            if (item->second.size() == n - 1 && trusts[item->first].size() == 0) {
                return item->first;
            }
        }
        return -1;
    }
};